src/controller/python/chip/yaml/YamlParser.py

```diff
@@ -0,0 +1,374 @@
+
+            # Create a cluster object for the request from the provided YAML data.
+            self._request_object = attribute(request_data)
+        else:
+            raise UnexpectedParsingError(f'WriteAttribute action does have arguments {self.label}')
+
+    def run_action(self, dev_ctrl: ChipDeviceCtrl, endpoint: int, node_id: int):
+        try:
+            resp = asyncio.run(
+                dev_ctrl.WriteAttribute(node_id, [(endpoint, self._request_object)]))
+        except chip.interaction_model.InteractionModelError:
+            if (self.expected_raw_response is not None and
+                    self.expected_raw_response.get('error')):
+                logger.debug('Got error, but was expected')
+            else:
+                raise
+
+        # TODO: confirm resp give a Success value, although not all write action are expected
+        # to succeed, hence why this is a todo and not simply just done. Below is example of
+        # what success check might look like.
+        # asserts.assert_equal(resp[0].Status, StatusEnum.Success, 'label write must succeed')
+
+
+class YamlTestParser:
+    '''Parses the test YAMLs and converts to a more natural Pythonic representation.
+
+    The parser also permits execution of those tests there-after.
+    '''
+
+    def __init__(self, yaml_path: str):
+        '''Constructor that parser the given a path to YAML test file.'''
+        with open(yaml_path, 'r') as stream:
+            try:
+                self._raw_data = yaml.safe_load(stream)
+            except yaml.YAMLError as exc:
+                raise exc
+
+        self._name = self._raw_data['name']
+        self._node_id = self._raw_data['config']['nodeId']
+        self._cluster = self._raw_data['config']['cluster'].replace(' ', '')
+        self._endpoint = self._raw_data['config']['endpoint']
+        self._base_action_test_list = []
+        self._data_model_lookup = PreDefinedDataModelLookup()
+
+        for item in self._raw_data['tests']:
+            # We only support parsing invoke interactions. As support for write/reads get added,
+            # these skips will be removed.
+            action = None
+            cluster = self._cluster
+            # Some of the tests contain 'cluster over-rides' that refer to a different
+            # cluster than that specified in 'config'.
+            if (item.get('cluster')):
+                cluster = item.get('cluster').replace(' ', '').replace('/', '')
+
+            if item.get('disabled'):
+                logger.info(f"Test is disabled, skipping {item['label']}")
+                continue
+            if item['command'] == 'writeAttribute':
+                action = self._attribute_write_action_factory(item, cluster)
+            elif item['command'] == 'readAttribute':
+                action = self._attribute_read_action_factory(item, cluster)
+            else:
+                action = self._invoke_action_factory(item, cluster)
```

Comment:
Does this properly handle the simulated cluster commands?

Reply:
Spoke with Cecille about this offline. Plan for bring in simulated cluster in the future is acceptable